test/plugin/test_in_wmi.rb

```diff
@@ -6,11 +6,49 @@ class WmiInputTest < Test::Unit::TestCase
     Fluent::Test.setup
   end
 
+  CONF = %[
+    tag monitor.wmi
+  ]
+
+  sub_test_case 'configuration' do
+    test 'valid class_name value' do
+      d = create_driver CONF + %[
+        class_name Win32_Processor
+      ]
+      assert_equal('Win32_Processor', d.instance.class_name)
+    end
+
+    test 'invalid class_name value' do
+      assert_raise(Fluent::ConfigError) {
```

Comment:
It would be better that it checks also the error message like this:

https://github.com/fluent/fluentd/blob/d7cebc6f3e0d338db50e5d099afe033273af5092/test/config/test_configurable.rb#L476

Reply:
Looks good, I'll try to rewrite later. Thank you!
Is there documentation available for Fluent::Configurable or Fluent::ConfigurableSpec?

Reply:
Sorry, above example is misleading.
Example for plugins: https://github.com/fluent/fluentd/blob/d7cebc6f3e0d338db50e5d099afe033273af5092/test/plugin/test_in_tcp.rb#L52

> Fluent::Configurable or Fluent::ConfigurableSpec?

Probably there is no document for themselves, and I think you don't need to know the detail for developing plugins.
[`Flunet::Configurable`](https://github.com/fluent/fluentd/blob/d7cebc6f3e0d338db50e5d099afe033273af5092/lib/fluent/configurable.rb#L25) is included by [`Flunet::Plugin::Base`](https://github.com/fluent/fluentd/blob/d7cebc6f3e0d338db50e5d099afe033273af5092/lib/fluent/plugin/base.rb#L24). It provides config related APIs like `config_param` to all plugins. The APIs are documented in the following pages:

* https://docs.fluentd.org/plugin-development/api-plugin-base#class-methods
* https://docs.fluentd.org/plugin-development/api-config-types

[`ConfigurableSpec`](https://github.com/fluent/fluentd/blob/d7cebc6f3e0d338db50e5d099afe033273af5092/test/config/test_configurable.rb#L6) is a module only for testing [`Flunet::Configurable`](https://github.com/fluent/fluentd/blob/d7cebc6f3e0d338db50e5d099afe033273af5092/lib/fluent/configurable.rb#L25), so you don't need to know until you modify Fluentd core.

Reply:
Oh I see. I understand that it is an indication to have to write a test including the details of the exception.